src/generic/toolbkg.cpp

```diff
@@ -358,26 +376,50 @@ bool wxToolbook::EnablePage(wxWindow *page, bool enable)
     return EnablePage(pageIndex, enable);
 }
 
+int wxToolbook::PageToToolId(size_t page) const
+{
+    wxASSERT_MSG(page < GetPageCount(), "Invalid page number");
+    return GetPage(page)->GetId();
+}
+
+int wxToolbook::ToolIdToPage(int toolId) const
+{
+    for (size_t i = 0; i < m_pages.size(); i++)
+    {
+        if (m_pages[i]->GetId() == toolId)
+        {
+            return (int) i;
+        }
+    }
+    return -1;
+}
+
 // ----------------------------------------------------------------------------
 // wxToolbook events
 // ----------------------------------------------------------------------------
 
 void wxToolbook::OnToolSelected(wxCommandEvent& event)
 {
-    const int selNew = event.GetId() - 1;
+    // find page for the tool
+    int page = ToolIdToPage(event.GetId());
+    if (page < 0)
```

Comment:
I think we should be more explicit and write `if ( page == wxNOT_FOUND )` as any other negative value is unexpected.